src/main/java/org/jboss/aerogear/unifiedpush/DefaultPushSender.java

```diff
@@ -243,9 +244,20 @@ protected String buildUrl() {
 
     @Override
     public void send(UnifiedMessage unifiedMessage, MessageResponseCallback callback) {
-        String jsonString = unifiedMessage.getObject().toJsonString();
+        final String jsonString = unifiedMessage.getObject().toJsonString();
         // fire!
-        submitPayload(buildUrl(), pushConfiguration.getConnectionSettings(), jsonString, pushConfiguration.getPushApplicationId(), pushConfiguration.getMasterSecret(), callback, new ArrayList<String>());
+        submitPayload(buildUrl(), pushConfiguration.getConnectionSettings(), jsonString, pushConfiguration.getPushApplicationId(), pushConfiguration.getMasterSecret(), callback, new ArrayList<>());
+    }
+
+    @Override
+    public void send(List<UnifiedMessage> unifiedMessages, MessageResponseCallback callback) {
+
+        final String jsonString = unifiedMessages.stream()
+                .map(unifiedMessage -> unifiedMessage.getObject().toJsonString())
```

Comment:
If we had a method that wraps `getObject().toJsonString()` you could even have used a method reference instead of a lambda ;) `UnifiedMessage::getJsonObjectString` for instance.